libs/blockchain-watcher/feature-app/src/lib/service/deliverService.ts

```diff
@@ -0,0 +1,113 @@
+import fetch from 'node-fetch'
+import { SlmChargeback, SlmEscrow, SlmPreorder } from '@solomon/shared/util-contract'
+import { envStore } from '../store/envStore'
+import { PaymentCreatedEvent } from '../Klass/PaymentCreatedEvent'
+import { stringHelper } from '../helper/stringHelper'
+import { EnumPaymentCreatedEventType } from '../Enum/EnumPaymentCreatedEventType'
+
+export class DeliverService {
+  async sendChargebackEvent(slmChargeback: SlmChargeback) {
+    const party1 = await slmChargeback.buyer()
+    const party2 = await slmChargeback.merchant()
+    const judge = await slmChargeback.judge()
+    const token = await slmChargeback.token()
+    const discount = await slmChargeback.discount()
+
+    const event = new PaymentCreatedEvent({
+      type: EnumPaymentCreatedEventType.chargeback,
+      party1: party1,
+      party2: party2,
+      contract: '', // todo
+      judgeContract: judge,
+      token: token,
+      discount: discount,
+      ethPaid: 0, // todo
+    })
+
+    const body = JSON.stringify(event)
+    const signature = stringHelper.generateDisputeApiSignature(
+      envStore.disputeApiSecretKey,
+      body,
+    )
+
+    await fetch(`${envStore.disputeApiUrl}/api/event`, {
+      method: 'post',
+      headers: {
+        'Content-type': 'application/json',
+        'X-Signature': signature,
+      },
+      body: body,
+    })
+  }
+
+  async sendPreorderEvent(slmPreorder: SlmPreorder) {
+    const party1 = await slmPreorder.buyer()
+    const party2 = await slmPreorder.merchant()
+    const judge = await slmPreorder.judge()
+    const token = await slmPreorder.token()
+    const discount = await slmPreorder.discount()
+
+    const event = new PaymentCreatedEvent({
+      type: EnumPaymentCreatedEventType.preorder,
+      party1: party1,
+      party2: party2,
+      contract: '', // todo
+      judgeContract: judge,
+      token: token,
+      discount: discount,
+      ethPaid: 0, // todo
+    })
+
+    const body = JSON.stringify(event)
+    const signature = stringHelper.generateDisputeApiSignature(
+      envStore.disputeApiSecretKey,
+      body,
+    )
+
+    await fetch(`${envStore.disputeApiUrl}/api/event`, {
+      method: 'post',
+      headers: {
+        'Content-type': 'application/json',
+        'X-Signature': signature,
+      },
+      body: body,
```

Comment:
can be written as `body,`

Reply:
Ah, I don't like writing `foo: foo` as `foo`. `foo: foo` is more clear and better when rename.

Reply:
VSCode will automatically detect and rename symbols, so I never had this problem.

I disagree but will not enforce this as a requirement.